src/app/project/qgsprojectelevationsettingswidget.cpp

```diff
@@ -241,6 +244,19 @@ bool QgsProjectElevationSettingsWidget::validate()
       mMessageBar->pushMessage( tr( "An elevation layer must be selected for a mesh terrain" ), Qgis::MessageLevel::Critical );
     }
   }
+
+  // show an error message if the lower value is greater than the upper one
+  // also show the error message if one of the values is not set
+  // however, do not show the error message if neither is set
+  const bool elevationLowerIsNotSet = mElevationLowerSpin->value() == mElevationLowerSpin->clearValue();
+  const bool elevationUpperIsNotSet = mElevationUpperSpin->value() == mElevationUpperSpin->clearValue();
```

Comment:
I think we'd benefit from a QgsDoubleSpinBox::isCleared / isClearValue

Reply:
Done.